packages/react-dom/src/__tests__/ReactDOM-test.js

```diff
@@ -380,6 +380,7 @@ describe('ReactDOM', () => {
   it('throws in DEV if jsdom is destroyed by the time setState() is called', () => {
     class App extends React.Component {
       state = {x: 1};
+      componentDidUpdate() {}
```

Comment:
What is this for?

Reply:
There's a comment below. It's used to trigger `invokeGuardedCallback`, which no longer wraps the begin phase unless an error is thrown.